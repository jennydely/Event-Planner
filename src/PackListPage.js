import React from 'react';
import styled from 'styled-components/macro'
import { NavLink, useParams } from 'react-router-dom'
import useEventForm from './componentsEventPage/useEventForm'
import usePacklists from './usePacklists';
import ListItem from '../src/common/ListItem'

export default function PackListPage() {
    const { packlistName } = useParams()
    const { packlists } = usePacklists()
    const chosenPacklist = packlists.find(packlist => packlist.name === '' || packlist.name === packlistName)
    const { goBack } = useEventForm()
    return (        <>
            <main>

                <ul> {packlistName ? <h1>Packlist - {packlistName}</h1> : <h1>No Packlist</h1>}

                    {chosenPacklist ? chosenPacklist.packlist.sort().map(item => (
                        <ItemStyled key={item}><input type="checkbox" />{item}</ItemStyled>
                    ))
                        : <NoPacklistText>There is no packlist added to this event</NoPacklistText>}
                </ul>

            </main>
            <footer>
                <NavLink to="/">
                    <button type="button" onClick={goBack}>Back</button>
                </NavLink>
            </footer>
        </>
    )
}
const NoPacklistText = styled(ListItem)`
margin-top: 200px;
`

const ItemStyled = styled(ListItem)`
`