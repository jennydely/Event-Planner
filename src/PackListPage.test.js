import React from 'react'
import { Router, Route } from 'react-router-dom'
import { render } from '@testing-library/react'
import { createMemoryHistory } from 'history'
import renderer from 'react-test-renderer'
import PackListPage from './PackListPage'
import 'jest-styled-components'

const renderWithRouter = (component) => {
    const history = createMemoryHistory()
    history.push('/packlist/festival')
    return { 
    ...render (
    <Router history={history}>
        <Route path='/packlist/:packlistName?'>

            {component}
        </Route>
    </Router>
    )
  }
}

describe('PackList', () => {
    const packlistName = "festival";
    const packlists = {
        "id": 500,
        "name": "festival",
        "packlist": 
           ["Wasser",
            "Plastikzelt",
            "Campingkocher",
            "Essen",
            "Pfanne & Öl",
            "Cola",
            "Festivalbox I",
            "Festivalbox II",
            "Ticket",
            "Waschschale",
            "Teppich",
            "Isomatte",
            "Luftmatratze",
            "Bettdecke & Kissen",
            "Knicklichter",
            "Alu-Herringe",
            "Powerbank & Ladekabel",
            "Kopftuch",
            "Portmornai mit Bargeld",
            "Kulturtasche",
            "Handtücher",
            "Kleidung",
            "Cider",
            "Metalgürtel",
            "ausgedruckter Timetable"
        ]}

    it('displays the packlist festival', () => {
        const { getByText } = renderWithRouter(<PackListPage packlists={packlists} packlistName={packlistName} />)
        setTimeout(() => {
            expect(getByText('festival')).toBeInTheDocument()
            expect(getByText(packlists[0].name)).toBeInTheDocument()
            expect(getByText(packlists[0].packlist)).toBeInTheDocument()
        }, 100)
    })
it('renders correctly', () => {
    const tree = renderer.create(<PackListPage packlists={packlists} packlistName={packlistName} />)
    expect(tree).toMatchSnapshot()
})
})