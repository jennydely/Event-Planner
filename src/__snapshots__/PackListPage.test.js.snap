// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PackList renders correctly 1`] = `
Array [
  .c0 {
  height: 92vh;
}

.c1 {
  text-align: center;
}

<main
    className="c0"
  >
    <h1>
      No PackList
    </h1>
    
    <p
      className="c1"
    >
      There is no packlist added to this event
    </p>
  </main>,
  <footer>
    <button
      onClick={[Function]}
      type="button"
    >
      Back
    </button>
  </footer>,
]
`;
